src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/LeoAuthProviderHelper.scala

```diff
@@ -44,7 +52,7 @@ class LeoAuthProviderHelper(wrappedAuthProvider: LeoAuthProvider, authConfig: Co
     }
 
     // recover from failed futures AND catch thrown exceptions
-    try { future.recoverWith(exceptionHandler) } catch exceptionHandler
+    try { future.withTimeout(providerTimeout, s"LeoAuthProvider timed out after $providerTimeout").recoverWith(exceptionHandler) } catch exceptionHandler
```

Comment:
Note the timeout is applied before the recovery, so a `TimeoutException` is mapped to a `LeoException` which is handled by our app!

Reply:
Also note: `withTimeout` comes from here: https://github.com/broadinstitute/workbench-libs/blob/develop/util/src/main/scala/org/broadinstitute/dsde/workbench/util/FutureSupport.scala#L51

Reply:
Just so I understand: the `TimeoutException` is turned into an `AuthProviderException` in the handler, right?

Or is the `TimeoutException` a subclass of `WorkbenchException` which is handled at the route-level exception handler?

Reply:
The former. So the `TimeoutException` and its message are thrown away and converted to an `AuthProviderException` here (which is handled upstream).

Just thinking, I could tweak the message here so users would see "provider timed out" instead of "provider errored" in this case.